lib/sycamore/sycamore/data/document.py

```diff
@@ -212,6 +212,25 @@ def __str__(self) -> str:
         }
         return json.dumps(d, indent=2)
 
+    def field_to_value(self, field: str) -> Any:
+        """
+        Extracts the value for a particular document field.
+
+        Args:
+            doc: The document
+            field: The field in dotted notation to indicate nesting, e.g. doc.properties.schema.
+
+        Returns:
+            The value associated with the document field.
+        """
+        fields = field.split(".")
+        value = getattr(self, fields[0])
+        if len(fields) > 1:
+            assert fields[0] == "properties"
```

Comment:
The idea is that it can support any field, e.g. "text_representation", "properties.entity.blah", "doc_id", etc. I think (?) that the only field that has multiple nested layers is properties, which is why the assert is there, but I could be wrong. Currently though, we don't have any custom error handling for this (it's Ray/python errors that show up for problems) — should I add anything?